cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go

```diff
@@ -23,119 +23,120 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// InitConfiguration contains a list of elements that is specific "kubeadm init"-only runtime
-// information.
+// InitConfiguration contains runtime information that are specific to "kubeadm init".
+// These information are only used the first time `kubeadm init` runs.
+// After that, the information in the fields IS NOT uploaded to the `kubeadm-config` ConfigMap
+// that is used by `kubeadm upgrade` for instance. These fields must be optional.
 type InitConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
-	// `kubeadm init`-only information. These fields are solely used the first time `kubeadm init` runs.
-	// After that, the information in the fields IS NOT uploaded to the `kubeadm-config` ConfigMap
-	// that is used by `kubeadm upgrade` for instance. These fields must be omitempty.
-
-	// BootstrapTokens is respected at `kubeadm init` time and describes a set of Bootstrap Tokens to create.
-	// This information IS NOT uploaded to the kubeadm cluster configmap, partly because of its sensitive nature
+	// `bootstrapTokens` describes a set of Bootstrap Tokens to create during `kubeadm init`.
+	// This information is NOT uploaded to the `kubeadm-config` ConfigMap, partly because of its sensitive nature.
 	BootstrapTokens []BootstrapToken `json:"bootstrapTokens,omitempty"`
 
-	// NodeRegistration holds fields that relate to registering the new control-plane node to the cluster
+	// `nodeRegistration` holds fields related to registering the new control-plane node to the cluster.
 	NodeRegistration NodeRegistrationOptions `json:"nodeRegistration,omitempty"`
 
-	// LocalAPIEndpoint represents the endpoint of the API server instance that's deployed on this control plane node
-	// In HA setups, this differs from ClusterConfiguration.ControlPlaneEndpoint in the sense that ControlPlaneEndpoint
-	// is the global endpoint for the cluster, which then loadbalances the requests to each individual API server. This
-	// configuration object lets you customize what IP/DNS name and port the local API server advertises it's accessible
-	// on. By default, kubeadm tries to auto-detect the IP of the default interface and use that, but in case that process
-	// fails you may set the desired value here.
+	// `localAPIEndpoint` represents the endpoint of the API server instance that's deployed on this control plane node.
+	// In HA setups, this differs from `ClusterConfiguration.controlPlaneEndpoint` in the sense that
+	// `controlPlaneEndpoint` is the global endpoint for the cluster, which loadbalances the requests to each individual
```

Comment:
it's not a common practice in k8s core APIs to surround the field names in ``` ` ```
is this something that:
- is present in other go docs out there?
- does it look nicer? (i'm yet to run this DIFF locally in a godoc server).

Reply:
There is no consensus on this, to be honest.
I'm leaning towards a consistent representation for field names. There are actually not too many choices and the problem isn't about whether the output looks nicer. It is really about highlighting. Instead of saying this:

> The port is defaulted to 6443.

We may want to say:

> The `port` is defaulted to 6443.

If we don't surround the field names with backtiqs, the sentence may be confusing.

We used to rely on CamelCase or PascalCase for resource kinds and we still do that in the website. But as we can see, field names start will lower case letters. It is less distinguishable if we don't highlight it.
This practice is applied **inconsistently** in website.

Reply:
like mentioned [here](https://github.com/kubernetes/kubernetes/pull/96215#discussion_r518089240) i think we should use PascalCase everywhere.

WRT quoting, the Kubernetes core API docs do not quote other fields (verified [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/)  and this would be my preference for the kubeadm / kubelet docs too. these two are considered ComponentConfig grade and not core APIs, but i think we should try to be consistent.